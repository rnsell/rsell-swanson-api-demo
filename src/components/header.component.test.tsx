import React from "react";
import { mount } from "enzyme";
import { Header, TotalVotes } from "./header.component";

describe("header.component.ts", () => {
  let component: any;

  beforeEach(() => {
    component = mount(<Header votes={15} />);
  });

  it("Should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should display correct number of votes.", () => {
    const totalVotes = component.find(TotalVotes);

    expect(totalVotes.text()).toBe(`Total Votes: ${15}`);
  });
});
