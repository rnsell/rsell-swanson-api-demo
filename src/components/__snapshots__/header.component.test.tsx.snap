// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header.component.ts Should match snapshot 1`] = `
<Header
  votes={15}
>
  <Styled(Row)
    data-testid="header__menu"
  >
    <Row
      className="sc-bdfBwQ jVopyv"
      data-testid="header__menu"
    >
      <div
        className="ant-row sc-bdfBwQ jVopyv"
        data-testid="header__menu"
        style={
          Object {
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
          }
        }
      >
        <Col
          xs={16}
        >
          <div
            className="ant-col ant-col-xs-16"
            style={Object {}}
          >
            <Row>
              <div
                className="ant-row"
                style={
                  Object {
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
              >
                <styled.h3
                  data-testid="header__title"
                >
                  <h3
                    className="sc-gsTCUz fRZIZs"
                    data-testid="header__title"
                  >
                    Ron Swanson Quote Voter
                  </h3>
                </styled.h3>
                <styled.span
                  data-testid="header__sub-title"
                >
                  <span
                    className="sc-dlfnbm bscQXW"
                    data-testid="header__sub-title"
                  >
                    "Vote for your favorite quote!"
                  </span>
                </styled.span>
              </div>
            </Row>
          </div>
        </Col>
        <Col
          xs={8}
        >
          <div
            className="ant-col ant-col-xs-8"
            style={Object {}}
          >
            <Styled(Row)
              data-testid="header__total_votes"
              justify="end"
            >
              <Row
                className="sc-hKgILt"
                data-testid="header__total_votes"
                justify="end"
              >
                <div
                  className="ant-row ant-row-end sc-hKgILt"
                  data-testid="header__total_votes"
                  style={
                    Object {
                      "marginBottom": undefined,
                      "marginLeft": undefined,
                      "marginRight": undefined,
                      "marginTop": undefined,
                    }
                  }
                >
                  Total Votes: 
                  15
                </div>
              </Row>
            </Styled(Row)>
          </div>
        </Col>
      </div>
    </Row>
  </Styled(Row)>
</Header>
`;
